website/docs/r/rde.html.markdown

```diff
@@ -0,0 +1,198 @@
+  name        = "My custom RDE"
+  resolve     = true
+  # Functions are evaluated eagerly, so the file must exist and not be a reference to a created Terraform resource
+  input_entity = templatefile("${path.module}/entities/custom-rde.json", {
+    name          = var.name
+    custom_field  = "This one is hardcoded"
+    another_field = var.anoter_field
+    replicas      = 2
+  })
+}
+
+output "computed_rde" {
+  value = vcd_rde.my_rde.computed_entity
+}
+```
+
+## Example Usage with a URL that contains a schema file
+
+```hcl
+data "vcd_rde_type" "my_type" {
+  vendor    = "bigcorp"
+  namespace = "tech1"
+  version   = "1.2.3"
+}
+
+resource "vcd_rde" "my-rde" {
+  org         = "my-org"
+  rde_type_id = data.vcd_rde_type.my-type.id
+  name        = "My custom RDE"
+  resolve     = true
+  entity_url  = "https://just.an-example.com/entities/custom-rde.json"
+}
+```
+
+## Input entity vs Computed entity
+
+There is a common use case for RDEs where they are used by 3rd party components that perform continuous updates on them,
+which are expected and desired. This conflicts with Terraform way of working, as doing a `terraform apply` would then
+perform actions to remove every single change done by those 3rd party tools, which we don't want in this case.
+
+To add compatibility with this scenario, there are two important arguments, `input_entity`/`input_entity_url`,
+and two important computed attribute, `computed_entity` and `entity_in_sync`.
+
+If your RDE is intended to be managed **only and exclusively** by Terraform, the contents of the input JSON should
+always match with those retrieved into `computed_entity`, and this will be reflected in the `entity_in_sync` attribute,
+which should be always `true`.
+
+Otherwise, only `computed_entity` will reflect the current state of the RDE in VCD and `entity_in_sync` will be `false`, whereas
+`input_entity` and `input_entity_url` will only specify the RDE contents that were used either on creation or in a deliberate
+update that will cause the RDE contents to be **completely overridden**.
+
+As per this last point, one needs to be careful when updating `input_entity` or `input_entity_url`, as Terraform will apply
+whatever changes were done, ignoring the real state from `computed_entity`. To perform a real update, one
+needs to check the contents of the `computed_entity` and do some diff with the original input.
+
+In other words:
+
+~> When you want to update an RDE and `entity_in_sync` is `false`, you should always merge the contents
+of `computed_entity` and `input_entity` to avoid overriding the whole entity by mistake with an old value. 
+
+## RDE resolution
+
+When a RDE is created, its `state` will be `PRE_CREATED`, which means that the entity JSON was not validated against the
+[RDE Type](/providers/vmware/vcd/latest/docs/resources/rde_type) schema. After resolution, `state` should be either `RESOLVED`
+or `RESOLUTION_ERROR` if the input JSON doesn't match the schema.
```

Comment:
I see. Reading the paragraph below one question comes up:

> The RDE must be resolved at some point to be used or deleted, and this operation can be done either by Terraform with `resolve=true`, or by a 3rd party actor that will do it behind the scenes (`resolve=false`).

1. If there's a schema mismatch, why can we mark it resolved? Doesn't it counter the purpose of the field?
2. If the schema was matched but later, does resolve become `true` automatically? Does Terraform pick that up?

Reply:
1) You can mark it resolved as you need to validate it against the schema in every case. So maybe there's a schema mismatch but you are not aware of it (it's not like you do resolve it deliberately knowing it has errors, it's more that a mistake/typo can happen and resolution would fail). In other words, you don't know there's a schema mismatch until you resolve it and get the feedback from VCD (we can think that "resolve"="validate" and "resolved"="validated", but I guess it's was not named like that for a good reason).

2) That's a good point. Right now, it doesn't, as it is acting more like an "imperative" argument rather than a "desired state" argument. The "state" in this case would be the purely Computed attribute `state`, which will be `RESOLVED` if it's resolved, even if this sort of imperative argument `resolve` was `false` (this case corresponds to an external actor who resolved it outside Terraform).

Not sure if I explained myself correctly, let me know if I can clarify if not.

Reply:
Thank you for an in depth explanation, I think it's good. @dataclouder @Didainius @adezxc is the "resolve" keyword and the whole resolution workflow clear enough from your perspective?

Reply:
good for me as well